input/script_utils.go

```diff
@@ -141,6 +143,40 @@ func GenFundingPkScript(aPub, bPub []byte, amt int64) ([]byte, *wire.TxOut, erro
 	return witnessScript, wire.NewTxOut(amt, pkScript), nil
 }
 
+// GenTaprootFundingScript constructs the taproot-native funding output that
+// uses musig2 to create a single aggregated key to anchor the channel.
+func GenTaprootFundingScript(aPub, bPub *btcec.PublicKey,
+	amt int64) ([]byte, *wire.TxOut, error) {
```

Comment:
From the code, it looks like the return triple is `pkScript, txOut, error`. Naive question, but is it well known that the `[]byte` type expressed in the return triple is the `pkScript`? Any benefit to readers to describe this in the function comment or can the return value be named to be more expressive?

Reply:
> Naive question, but is it well known that the []byte type expressed in the return triple is the pkScript

Not necessarily, this is a good point about the readability of this fragment. The reason the script is returned alongside here is that for p2wsh outputs, the witness script is also needed to sign new commitment updates (construct the sighash specifically). For these musig2+taproot channels, we sign new commitments using the keyspend path, so only output script is returned. 

In this case, since we're already returning the `TxOut`, we actually don't need to also return the `pkScript`. I think I did that initially to make things symmetric, but it isn't necessary.